providers/implementations/ciphers/cipher_null.c

```diff
@@ -82,7 +82,7 @@ static int null_cipher(void *vctx, unsigned char *out, size_t *outl,
     }
     if (outsize < inl)
         return 0;
-    if (in != out)
+    if (out != NULL && in != out)
```

Comment:
I'd prefer `inl != 0 && !ossl_assert(out != NULL) && in != out` - please verify that it was called with inl == 0 when it was crashing.  If not, that is a bug at the caller.

Reply:
It crashes if this is used as AEAD cipher and `EVP_CipherUpdate` is called for AAD (additional authenticated data) where `out == NULL` and `inl != 0`.

Reply:
We are not using it in integrity only ciphers for TLS 1.3. I could just remove this modification without any issue.

But if it doesn't create any issue, then it will be nice that null cipher can also be used as other ciphers without any specific consideration.

Reply:
OK then